### PR TITLE
Added plot state qsphere code example to API reference in order to better showcase the different arguments

### PR DESCRIPTION
**Pull request by Guillermo-Mijares-Vilarino**
### Summary
Added a new `plot_state_qsphere` code example to the [API reference](https://qiskit.org/documentation/stubs/qiskit.visualization.plot_state_qsphere.html#qiskit.visualization.plot_state_qsphere) to better showcase the different arguments it has.

### Details and comments
The new example uses the `show_state_phases` and `use_degrees` arguments to show the phases in degrees instead of radians and uses a `DensityMatrix` instead of a `Statevector`. The circuit is also a bit more complex so the different colors and point sizes can be seen.

### CODE REVIEW

File: qiskit/visualization/state_visualization.py [hunk 1]
```diff
@@ -710,8 +710,26 @@ def plot_state_qsphere(
            qc.h(0)
            qc.cx(0, 1)
 
-           state = Statevector.from_instruction(qc)
+           state = Statevector(qc)
            plot_state_qsphere(state)
+        .. jupyter-execute::
+
+           from qiskit import QuantumCircuit
```
Comment (mtreinish) [inline qiskit/visualization/state_visualization.py:717]:
```suggestion
           state = Statevector(qc)
           plot_state_qsphere(state)
           
        .. jupyter-execute::

           from qiskit import QuantumCircuit
```

File: qiskit/visualization/state_visualization.py [hunk 2]
```diff
@@ -710,8 +710,26 @@ def plot_state_qsphere(
...
+           from qiskit.quantum_info import DensityMatrix
+           from qiskit.visualization import plot_state_qsphere
+           import numpy as np
+           %matplotlib inline
```
Comment (mtreinish) [inline qiskit/visualization/state_visualization.py:721]: I don't think this is needed the jupyter execute directives in a document share a kernel. (also I'm not sure we need it generally, but regardless just having a single magic in the first cell in the page is sufficient)

Comment (HuangJunye) [reply qiskit/visualization/state_visualization.py:721]: I am not sure %matplotlib inline is needed either. @Guillermo-Mijares-Vilarino can you try to remove this and see if the plots are still rendered as expected?

Comment (Guillermo-Mijares-Vilarino) [reply qiskit/visualization/state_visualization.py:722]: Sure!

Comment (Guillermo-Mijares-Vilarino) [reply qiskit/visualization/state_visualization.py:722]: Removed `%matplotlib inline` and the docs were built successfully.

File: qiskit/visualization/state_visualization.py [hunk 3]
```diff
@@ -698,20 +698,35 @@ def plot_state_qsphere(
...
+
+           qc = QuantumCircuit(2)
+           qc.h([0, 1])
+           qc.cz(0,1)
+           qc.ry(np.pi/3, 0)
+           qc.rx(np.pi/5, 1)
+           qc.z(1)
+
+           matrix = DensityMatrix(qc)
+           plot_state_qsphere(matrix, figsize = (13,9),
```
Comment (HuangJunye) [inline qiskit/visualization/state_visualization.py:728]: I think we should keep the `figsize` to the module level documentation.

**Review by HuangJunye — APPROVED** [review]
LGTM! Thanks for making this contribution.

**Review by 1ucian0 — APPROVED** [review]
Thanks!

**Pull request merged**